packages/cardpay-sdk/sdk/token-bridge-foreign-side.ts

```diff
@@ -40,26 +43,41 @@ const CLAIM_BRIDGED_TOKENS_GAS_LIMIT = 290000;
 export default class TokenBridgeForeignSide implements ITokenBridgeForeignSide {
   constructor(private layer1Web3: Web3) {}
 
+  async unlockTokens(txnHash: string): Promise<TransactionReceipt>;
   async unlockTokens(
     tokenAddress: string,
     amount: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async unlockTokens(
+    tokenAddressOrTxnHash: string,
+    amount?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    if (!amount) {
```

Comment:
ditto with the comments above about testing whether the value you actually care about is shaped like an address or txnHash instead of testing a side effect